packages/webcert/src/components/header/WebcertHeaderUnit.tsx

```diff
@@ -24,25 +36,104 @@ const InactiveUnit = styled.span`
+  position: absolute;
+  background: #ffffff;
+  padding: 1.2em 2em;
+  box-shadow: 1px 1px 10px #ccc;
+  right: -11px;
+  margin-top: 0.5em;
+  border-radius: 3px;
+  z-index: 9999;
+
+  &::before {
+    content: '';
+    width: 10px;
+    height: 10px;
+    background: #ffffff;
+    transform: rotate(45deg);
+    position: absolute;
+    top: -5px;
+    right: 12px;
+  }
+`
+
+const RelativeDiv = styled.div`
+  position: relative;
+`
+
 const WebcertHeaderUnit: React.FC = () => {
+  const dispatch = useDispatch()
   const user = useSelector(getUser, shallowEqual)
+  const totalDraftsAndUnhandledQuestionsOnOtherUnits = useSelector(getTotalDraftsAndUnhandledQuestionsOnOtherUnits)
+  const userLinks = useSelector(getUserResourceLinks)
+  const [isExpanded, setIsExpanded] = useState(false)
+
+  const changeUnitLink = userLinks?.find((link) => link.type === ResourceLinkType.CHANGE_UNIT)
+
+  const toggleMenu = () => {
+    setIsExpanded(!isExpanded)
+  }
+
+  const expandButton = () => {
+    if (changeUnitLink) {
+      return (
+        <StyledButton onClick={toggleMenu} tabIndex={0} className="iu-ml-300" data-testid="expandChangeUnit">
+          {isExpanded ? (
+            <ArrowUp src={arrow} alt="" data-testid="expandArrow" />
+          ) : (
+            <ArrowDown src={arrow} alt="" data-testid="expandArrow" />
+          )}
+        </StyledButton>
+      )
+    }
+  }
+
+  const openModal = () => {
+    if (changeUnitLink) {
+      dispatch(updateIsCareProviderModalOpen(true))
+      setIsExpanded(false)
+    }
+  }
 
   const toString = (user: User): React.ReactNode => {
     return (
-      <Wrapper>
-        <span>
-          {user.loggedInCareProvider.unitName} <br />
-          <Italic>{user.loggedInUnit.unitName}</Italic>
-        </span>
-        {user.loggedInUnit.isInactive ? (
-          <InactiveUnit
-            className="iu-ml-400"
-            data-tip="Enheten är markerad som inaktiv i journalsystemet, vilket innebär att viss funktionalitet ej är tillgänglig.">
-            <AlertCircle />
-            <span>Inaktiv enhet</span>
-          </InactiveUnit>
-        ) : null}
-      </Wrapper>
+      <RelativeDiv>
+        <Wrapper>
+          <span>
+            {user.loggedInCareProvider.unitName} - {user.loggedInUnit.unitName}
+            <br />
+            <Italic>
+              {changeUnitLink &&
+                `${totalDraftsAndUnhandledQuestionsOnOtherUnits} ej hanterade ärenden och ej signerade utkast på andra vårdenheter.`}
+            </Italic>
+          </span>
+          {user.loggedInUnit.isInactive ? (
+            <InactiveUnit
+              className="iu-ml-400"
+              data-tip="Enheten är markerad som inaktiv i journalsystemet, vilket innebär att viss funktionalitet ej är tillgänglig.">
+              <AlertCircle />
+              <span>Inaktiv enhet</span>
+            </InactiveUnit>
+          ) : null}
+          {expandButton()}
+        </Wrapper>
+        {isExpanded && (
```

Comment:
This should be extracted to its own component and the expanded button should be made general (see comment above)